lib/react.dart

```diff
@@ -565,12 +571,39 @@ abstract class Component2 implements Component {
   /// [setState]. The old props can be accessed via [props].
   ///
   /// Calling [setState] within this function will not trigger an additional [render].
+  /// > __DEPRECATED - DO NOT USE__
+  /// >
+  /// > This will be removed once 6.0.0 releases, switching to [Component2.getDerivedStateFromProps] is the path forward
+  /// >
+  /// > React 16 has deprecated [componentWillReceiveProps] in favor of [getDerivedStateFromProps]
+  /// >
+  /// > This will be completely removed when the JS side of it is slated for removal (ReactJS 17 / react.dart 6.0.0)
+  @mustCallSuper
+  @Deprecated('6.0.0')
+  void componentWillReceiveProps(Map newProps) {}
+
+  /// ReactJS lifecycle method that is invoked before rendering on the intial mount and on subsequent updates
+  /// It should return an object to update the state, or null to update nothing (this is different from componentWillRecieveProps).
   ///
-  /// __Note__: Choose either this method or [componentWillReceivePropsWithContext]. They are both called at the same
-  /// time so using both provides no added benefit.
+  /// [prevState] will be null when this lifcecyle method is called before first mount.
   ///
-  /// See: <https://facebook.github.io/react/docs/react-component.html#updating-componentwillreceiveprops>
-  void componentWillReceiveProps(Map newProps) {}
+  /// This method is also static, so features like [this] or [ref] cannot be used in this method
+  ///
+  /// See: <https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops>
```

Comment:
I think it'd be a good idea to include the info under the "Deriving state leads to verbose code and makes your components difficult to think about." section of these docs directly within this doc comment.